gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/v4/NativeApi.java

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactive.handlers.api.v4;
+
+import io.gravitee.definition.model.DefinitionVersion;
+import io.gravitee.definition.model.Policy;
+import io.gravitee.definition.model.v4.flow.Flow;
+import io.gravitee.definition.model.v4.flow.step.Step;
+import io.gravitee.definition.model.v4.nativeapi.NativeFlow;
+import io.gravitee.definition.model.v4.plan.AbstractPlan;
+import io.gravitee.definition.model.v4.plan.Plan;
+import io.gravitee.definition.model.v4.plan.PlanSecurity;
+import io.gravitee.definition.model.v4.resource.Resource;
+import io.gravitee.gateway.reactor.ReactableApi;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class NativeApi extends ReactableApi<io.gravitee.definition.model.v4.nativeapi.NativeApi> {
+
+    public NativeApi() {
+        super();
+    }
+
+    public NativeApi(io.gravitee.definition.model.v4.nativeapi.NativeApi nativeApi) {
+        super(nativeApi);
+    }
+
+    @Override
+    public String getApiVersion() {
+        return definition.getApiVersion();
+    }
+
+    @Override
+    public DefinitionVersion getDefinitionVersion() {
+        return definition.getDefinitionVersion();
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return definition.getTags();
+    }
+
+    @Override
+    public String getId() {
+        return definition.getId();
+    }
+
+    @Override
+    public String getName() {
+        return definition.getName();
+    }
+
+    @Override
+    public Set<String> getSubscribablePlans() {
+        return definition.getPlans().stream().filter(AbstractPlan::isSubscribable).map(AbstractPlan::getId).collect(Collectors.toSet());
+    }
+
+    @Override
+    public Set<String> getApiKeyPlans() {
+        return definition.getPlans().stream().filter(AbstractPlan::isApiKey).map(AbstractPlan::getId).collect(Collectors.toSet());
+    }
+
+    @Override
+    public <D> Set<D> dependencies(Class<D> type) {
+        if (Policy.class.equals(type)) {
+            return (Set<D>) policies();
+        } else if (Resource.class.equals(type) && definition.getResources() != null) {
+            return (Set<D>) new HashSet<>(definition.getResources());
+        }
+
+        return Collections.emptySet();
+    }
+
+    private Set<Policy> policies() {
+        Set<Policy> policies = new HashSet<>();
+
+        // Load policies from Plans
+        if (definition.getPlans() != null) {
+            definition
+                .getPlans()
+                .forEach(plan -> {
+                    // TODO: associate plan security with its policy (https://github.com/gravitee-io/issues/issues/8427)
+                    if (plan.useStandardMode()) {
```

Comment:
I don't remember what is standard mode and if it applies to native API?

Reply:
it is to differentiate from PUSH plans (webhook), so it does not apply here good catch